packages/ra-core/src/controller/details/index.ts

```diff
@@ -0,0 +1,25 @@
+import {
+    CreateControllerProps,
+    useCreateController,
+} from './useCreateController';
+import { EditControllerProps, useEditController } from './useEditController';
+import { ShowControllerProps, useShowController } from './useShowController';
+
+export * from './CreateBase';
+export * from './CreateContext';
+export * from './CreateContextProvider';
+export * from './CreateController';
+export * from './EditBase';
+export * from './EditContext';
+export * from './EditContextProvider';
+export * from './EditController';
+export * from './ShowBase';
+export * from './ShowContext';
+export * from './ShowContextProvider';
+export * from './ShowController';
+
+// We don't want to export CreateProps, EditProps and ShowProps as they should
```

Comment:
I think they're exported anyway because of the export * above

Reply:
They're not. Otherwise TS would have complained

Reply:
hmm I wouldn't trust TypeScript on this one

Reply:
And no, because they are declared in the files for which we don't have an `export *`